tests/agent_equipment/test_source_resolution.py

```
@@ -0,0 +1,1168 @@
+                {
+                    "kind": "git",
+                    "repository": "https://example.com/equipment.git",
+                },
+                {"all": True},
+                patterned_value("distribution:", over_limit),
+            ),
+        )
+        for source, selection, distribution_identity in cases:
+            with (
+                self.subTest(distribution_identity=distribution_identity),
+                self.assertRaisesRegex(ValueError, "bounded|string|invalid"),
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)